filters/special_casing/README.md

```diff
@@ -0,0 +1,22 @@
+## Special Casing filter
+
+## What type of a filter is this?
+
+This filter checks if the input sentence has a special casing, i.e. the string is either all lowercased, all uppercased or has title casing.
+Author: Vikas Raunak (viraunak@microsoft.com)
+
+## Why is measuring performance on this split important?
+Traditionally, neural models are trained without any explicit casing information provided to the tokenizer. The most widely used data-driven tokenization algorithms, such as sentencepiece (Unigram LM) or Byte-Pair-Encoding typically do not have any explicit factors to handle casing information. Therefore, tackling edge cases pertaining to case is important in text generation problems, e.g. in Machine Translation if a newspaper headline in title casing is provided to the model, the general expectation is that the output should preseve this title case. Same is true for inputs which are fully uppercased (typical in ALERT messages) or fully lowercased (typical in chat messages).
+
+This filter can be used to create splits of specially cased sentences of 3 types, fully uppercased, fully lowercased or titlecased. Filtering out and testing on examples belonging to such special cased inputs can provide feedback for improving the model or guiding data augmentation accordingly.
+
```

Comment:
Can you add a robustness evaluation on some of the standard dataset/model combinations? Check [this](https://github.com/GEM-benchmark/NL-Augmenter/tree/main/evaluation) for details.

Reply:
Thanks, I am not sure if standard dataset pre-processing already lowercases everything in these tasks. Do you suggest any particular dataset/task from above that reports numbers while preserving input casing?

Reply:
@richplant also the robustness-evaluation seems to be only for transformations, rather than filters, so skipping this as of now.